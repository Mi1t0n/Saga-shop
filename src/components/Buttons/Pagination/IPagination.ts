interface IPagination {
    getPage: (page: number) => void
    maxPage:number
}

export type {IPagination}