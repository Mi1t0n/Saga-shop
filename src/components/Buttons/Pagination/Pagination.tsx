import styled from "styled-components";
import {FC} from "react";
import {IPagination} from "./IPagination";

const Button = styled.button`
  border-radius: 50%;
  width: 25px;
  height: 25px;
  padding: 5px;
  border: 1px black solid;
  cursor: pointer;
`
const Wrapper = styled.div`
  align-items: center;
  display: flex;
  gap: 30px;
  justify-content: center;
`


const Pagination: FC<IPagination> = ({page, next, prev}) =>
    <Wrapper>
        <Button onClick={prev} data-testid='increase'>&lt;</Button>
        <h3 data-testid='page'>{page}</h3>
        <Button onClick={next} data-testid='decrease'>&gt;</Button>
    </Wrapper>

export default Pagination;
