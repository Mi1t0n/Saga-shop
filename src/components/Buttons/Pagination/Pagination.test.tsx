import Pagination from "./Pagination";
import {render, screen} from "@testing-library/react";
import React from "react";

describe('Pagination', () => {

    beforeEach(() => {
        render(
            <Pagination page={1} next={() => {
            }} prev={() => {
            }}/>
        )
    })

    it('Match snapshot', () => {
        const {asFragment} = render(<Pagination page={1} next={() => {
        }} prev={() => {
        }}/>)
        expect(asFragment()).toMatchSnapshot()
    })
    it('Initial page 1', () => {
        expect(screen.getByTestId('page')).toHaveTextContent('1')
    })
    it('Increase page', () => {
        expect(screen.getByTestId('increase')).toBeInTheDocument()
    })
    it('Decrease page', () => {
        expect(screen.getByTestId('decrease')).toBeInTheDocument()
    })

})