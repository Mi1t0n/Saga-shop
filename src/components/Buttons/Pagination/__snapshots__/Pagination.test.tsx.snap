// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gswNZR kvDuoZ"
  >
    <button
      class="sc-bcXHqe bVEkev"
      data-testid="increase"
    >
      &lt;
    </button>
    <h3
      data-testid="page"
    >
      1
    </h3>
    <button
      class="sc-bcXHqe bVEkev"
      data-testid="decrease"
    >
      &gt;
    </button>
  </div>
</DocumentFragment>
`;
